orchestrator/orchestrator.go

```diff
@@ -20,20 +20,50 @@ import (
 	executionmodule "github.com/mesg-foundation/engine/x/execution"
 	processmodule "github.com/mesg-foundation/engine/x/process"
 	runnermodule "github.com/mesg-foundation/engine/x/runner"
-	"github.com/sirupsen/logrus"
+	tmlog "github.com/tendermint/tendermint/libs/log"
 )
 
 func init() {
 	xrand.SeedInit()
 }
 
+const (
+	// AllLogTopic is the default topic where all logs are published to.
+	AllLogTopic = "*"
+)
+
+// Orchestrator manages the executions based on the definition of the processes
+type Orchestrator struct {
+	rpc             *cosmos.RPC
+	ep              *publisher.EventPublisher
+	eventStream     *event.Listener
+	executionStream chan *execution.Execution
+	stopC           chan bool
+	logs            *pubsub.PubSub
+	logger          tmlog.Logger
+	execPrice       string
+}
+
+// Log is representing a Log from the Orchestrator
+type Log struct {
```

Comment:
Might be a specific proto type to avoid the conversion and mistakes with the matching of parameters

Reply:
closing because same as https://github.com/mesg-foundation/engine/pull/1812#discussion_r422082793